cmd/start.go

```diff
@@ -91,3 +110,98 @@ func trapSignal(done func()) {
 	// call the cleanup func
 	done()
 }
+
+// UpdateClientsFromChains takes src, dst chains and update clients based on expiry time
+func UpdateClientsFromChains(src, dst *relayer.Chain) (err error) {
+	if src.PathEnd.ClientID == "" || dst.PathEnd.ClientID == "" {
+		return nil
+	}
+	eg := new(errgroup.Group)
+	eg.Go(func() error {
+		err = GetClientAndUpdate(src, dst)
+		return err
+	})
+	eg.Go(func() error {
+		err = GetClientAndUpdate(dst, src)
+		return err
+	})
+	err = eg.Wait()
+	return err
+}
+
+// GetClientAndUpdate update clients to prevent expiry
+func GetClientAndUpdate(src, dst *relayer.Chain) error {
+	height, err := src.QueryLatestHeight()
+	if err != nil {
+		return err
+	}
+
+	clientStateRes, err := src.QueryClientState(height)
+	if err != nil {
+		return err
+	}
+
+	// unpack any into ibc tendermint client state
+	clientStateExported, err := clienttypes.UnpackClientState(clientStateRes.ClientState)
+	if err != nil {
+		return err
+	}
+
+	// cast from interface to concrete type
+	clientState, ok := clientStateExported.(*ibctmtypes.ClientState)
+	if !ok {
+		return fmt.Errorf("error when casting exported clientstate on chain: %s", src.PathEnd.ClientID)
+	}
+
+	// query the latest consensus state of the potential matching client
+	consensusStateResp, err := clientutils.QueryConsensusStateABCI(src.CLIContext(0),
+		src.PathEnd.ClientID, clientState.GetLatestHeight())
+	if err != nil {
+		return err
+	}
+
+	exportedConsState, err := clienttypes.UnpackConsensusState(consensusStateResp.ConsensusState)
+	if err != nil {
+		return err
+	}
+
+	consensusState, ok := exportedConsState.(*ibctmtypes.ConsensusState)
+	if !ok {
+		return fmt.Errorf("error: consensus state is not tendermint type on chain %s", src.PathEnd.ChainID)
+	}
+
+	expirationTime := consensusState.Timestamp.Add(clientState.TrustingPeriod)
+
+	checkExpiryTime := viper.GetFloat64(flagThresholdTime)
+
+	// Checking expiration time is below 10minutes to current time
+	if expirationTime.After(time.Now()) && time.Until(expirationTime).Minutes() <= checkExpiryTime {
+		sh, err := relayer.NewSyncHeaders(src, dst)
+		if err != nil {
+			return err
+		}
+
+		dstUH, err := sh.GetTrustedHeader(dst, src)
+		if err != nil {
+			return err
+		}
+
+		msgs := []sdk.Msg{
+			src.UpdateClient(dstUH),
+		}
+
+		_, _, err = src.SendMsgs(msgs)
```

Comment:
I think we need to check the other boolean, if I remember correctly error are returned when the msg failed to send but there should be a boolean indicating if the message was successful